### PR TITLE
New and modified features (Notes colorizing, reminder, templates, todo lists, improved tag search, bug fixes)

### PR DESCRIPTION
**Pull request by antonkhvorov**
Greetings, Simplenote community!

Several weeks ago in our university on the Methods: Deciding what to Design course we were asked to find an open-source project on the Internet to implement, modify, and delete some features in it. And I am sure you already got the point - Simplenote became our choice.

Just a couple words about our course objectives: methods course is all about managing a project in the circumstances where requirements are changing. It focuses on requirements elicitation. In the beginning of the project, we have needed to find a person who will play customer role for us during the course. Our semester is coming to the end, so the final presentation will be on 21st of November. We would really appreciate if you as a Simplenote development team will provide us feedback about our work and it would be the best reward for our work if you will find some of these features useful and implement them in original Simplenote!

So, straight to the point - **changelog**:
## New features:
1. Notes Colorizing. Now you can select color for the note, which showing both in note editor and note list screen as a small vertical line aligned to the left side.

image

2. Templates. Create a template to make notes from it! It could be a template for shopping, for setting up the meetings and so on.

3. Todo lists. Now it is even easier to track your things done with Simplenote.

image

4. Reminders. Put a reminder on the note to not forget things you need to remember!
(Note: based on #381 PR)

image

## Modified features:
1. Search by tags. Now you can select multiple tags to increase search speed and find necessary information in a flash. The more tags you select, the more precise result will be.
image

## Bug fixes:
1. Fixed bug when tags were containing capital and/or different from latin letters and were not counting in tags editor.
2. Fixed bug with disappearing BottomSheetDialog (#394 issue here).

...and some minor changes!

We are still working on Simplenote, and right now we are developing an interactive tutorial. Feel free to visit our YouTrack:
http://youtrack.writeup.pro:8112/issues
and GitHub:
https://github.com/Asmadek/simplenote-android

We would really appreciate if you provide some comments and feedback about our work, with guiding us in the way to help improve Simplenote!

### CODE REVIEW

**Comment by beaucollins** [comment]
I love seeing this PR! It's going to be a hefty one to review.

Do you think it at all possible to separate the different features into individual pull requests?

**Comment by day7ok** [comment]
We love it being here too :)

Well, it would just take some time, I dont see any problems which could appear with separating this features to individual PR.
I guess that for the overview of the project you can use this one, and then tell us which one you like the most! After that, we can separate these features to individual PRs and focus on them.

File: README.md [hunk 1]
```diff
@@ -27,15 +29,23 @@ sdk.dir=/Applications/Android Studio.app/sdk
 Add your simperium appid and key to Simplenote/gradle.properties, and an empty googleAnalyticsId:
 
 ```
-simperiumAppId=SIMPERIUM_APP_ID
-simperiumAppKey=SIMPERIUM_KEY
+simperiumAppId=yield-cannons-00f
+simperiumAppKey=f4d9e4bab5fd40909481d91ae1f0d932
```
Comment (roundhill) [inline README.md]: @day7ok I've deleted this key since it is now compromised, you can generate a new one from your Simperium console.

**Comment by roundhill** [comment]
> I guess that for the overview of the project you can use this one, and then tell us which one you like the most! After that, we can separate these features to individual PRs and focus on them.

That might be good. What we typically do is evaluate if a feature is worth having in _all_ of the apps (iOS, web, electron) and then implement it across the board. Perhaps @drw158 could have a look at the features here to see if any of them are worth pursuing further?

File: README.md (hunk 1 above)
Comment (day7ok) [inline README.md]: @dmsnell @roundhill Sorry about that. We forgot to remove it since we have not used readme.md much

**Comment by day7ok** [comment]
@roundhill , @dmsnell , @beaucollins 
May I kindly ask you to make a short review about features in this PR, so we can make a separate PR for those in which you are interested in? Like, just to download and play with it for a while? That would really help us with our presentation in university, and then, to focus our effort on features which interested you the most.
Or should we wait until @drw158 will provide a feedback?

**Comment by dmsnell** [comment]
@day7ok @drw158 has (in my opinion) excellent intuition and skill in this area and I would trust his opinion far above my own.

this being said, I can say a few things about this. first of all, when we say you dump your PR we were all super excited and think this is an awesome project to work on for a project. your team did a remarkable job and some of these features are incredibly cool.

we tend to say "no" to most feature requests on Simplenote (even our own which we are excited about). as @roundhill mentioned we also try to make sure that all apps across the Simplenote spectrum can or will support the features we do let in. the simplicity of Simplenote is one of the things people say they love the most about it and this is one of the reasons we tend to guard it so much. I simply say this so that you do not feel discouraged if we do not want to move these features into the app.

so we're excited for you, think this is great, and look forward to see what the next steps will be. cheers and well done! 👍

**Comment by drw158** [comment]
I'll take a look asap

**Comment by drw158** [comment]
First off, thank you so much for contributing to Simplenote. It's a passion project for many of us, and we want it to keep going.

While going through new features I keep in mind: http://www.productstrategymeanssayingno.com

and 

> First and foremost, Simplenote is a premier note-taking application. Every design decision and feature addition leads to a better user experience that makes it easier to record thoughts and ideas. 

Disclaimer: I haven't actually haven't played with your PR. I'm just going off of the images.

Colorizing: I've been thinking a lot about this, but I'm still on the fence. I like the left-border implementation, but I'm not convinced this makes it easier to record thoughts and ideas. I think pinned notes satisfies the same need, but in a slightly different way. Because we already have a pin option, colorizing is unnecessary.

Templates: Very interesting idea! Again, I haven't used any of these new features, but I could see how this could be useful. I'll take a look.

Todo list: This has been something we have been wanting to do for a long time, and we need to do it right. Google Keep is an example of not doing it right imo. At least not right for us. One of the best implementations I've seen is by the Canvas iOS app. You can create a checklist item just by swiping. I'm not against doing todo lists first on Android, but if we do it, we need to do it across all apps in a short time frame. That means at least coming up with a design/plan for each platform.

Reminders: Also saw this feature on Google Keep. It's nice and I could see the use case, but I can't get behind it. I don't think it makes the core of Simplenote easier (recording thoughts and ideas quickly and simply).

Obviously exaggerated, but:
> Are you trying to become a “mobile collaboration tool that can send invoices, schedule tweets, and make coffee?” Optional features lead to bloatware, with no clear focus.

Multiple tag search: Seems useful. I'll need to use it to give an opinion. I think on desktop, maybe the user could just type in multiple tags in the search bar and do the same thing. On mobile, it's not easy typing that much, so I could see how selecting multiple tags could be beneficial.

**Comment by day7ok** [comment]
@drw158 Thank you so much for the detailed feedback.
I would like to say that our team was trying to keep Simplenote simple and have put decent effort to it. I believe that you will notice it when you will play with it, because some feedback can change during the actual application testing, e.g. as you said about **Tags Search** feature that it would not be convenient to use it due to writing it on the phone would be not so easy. But in reality - it's super easy since you need just to tap on the desirable tag on the menu to add it in the tag search panel.
Talking about **Colorizing** feature - we were trying to make it not only available for highlighting important notes (like pin feature), but also to provide user opportunity to have an associations with color and note content.
Once again, we were trying to keep it simple and friendly:)

**Comment by Jonatino** [comment]
Omg YES pls pls pls +1000000 👍

**Comment by CorentinLemaitre** [comment]
Just ++ 1000000  because even if it does not exist i am loving it.

**Comment by roundhill** [comment]
Closing for now since this went stale.

**Comment by GTD4life** [comment]
Love this approach , too!!!

**Pull request closed without merging**